### ISSUE TITLE
Failed to pull image networkservicemesh/nsm-coredns:latest

### ISSUE THREAD

**Issue by wjstk16** [networkservicemesh/networkservicemesh#1591]
There is no latest tag nsm images in docker hub.

```
nsm-system    alpine-nsc-5979b9ccc9-9lbcw                   1/3     ErrImagePull   0         106s
nsm-system    alpine-nsc-5979b9ccc9-hmwvx                   1/3     ErrImagePull   0        106s
nsm-system    alpine-nsc-5979b9ccc9-p2zk6                   1/3     ErrImagePull   0          106s
nsm-system    alpine-nsc-5979b9ccc9-snxm5                   1/3     ErrImagePull   0         106s
```

```
Events:
  Type     Reason     Age                From                  Message
  ----     ------     ----               ----                  -------
  Normal   Scheduled  99s                default-scheduler     Successfully assigned nsm-system/alpine-nsc-5979b9ccc9-9lbcw to kube-master
  Normal   Pulled     98s                kubelet, kube-master  Container image "networkservicemesh/nsm-init:latest" already present on machine
  Normal   Created    98s                kubelet, kube-master  Created container nsm-init-container
  Normal   Started    97s                kubelet, kube-master  Started container nsm-init-container
  Normal   Pulling    90s                kubelet, kube-master  Pulling image "alpine:latest"
  Normal   Created    84s                kubelet, kube-master  Created container alpine-img
  Normal   Pulled     84s                kubelet, kube-master  Successfully pulled image "alpine:latest"
  Normal   Started    84s                kubelet, kube-master  Started container alpine-img
  Warning  Failed     55s                kubelet, kube-master  Failed to pull image "networkservicemesh/nsm-monitor:latest": rpc error: code = Unknown desc = Error response from daemon: manifest for networkservicemesh/nsm-monitor:latest not found: manifest unknown: manifest unknown
  Warning  Failed     55s                kubelet, kube-master  Error: ErrImagePull
  Normal   Pulling    54s (x2 over 84s)  kubelet, kube-master  Pulling image "networkservicemesh/nsm-coredns:latest"
  Warning  Failed     47s (x2 over 75s)  kubelet, kube-master  Failed to pull image "networkservicemesh/nsm-coredns:latest": rpc error: code = Unknown desc = Error response from daemon: manifest for networkservicemesh/nsm-coredns:latest not found: manifest unknown: manifest unknown
  Warning  Failed     47s                kubelet, kube-master  Error: ImagePullBackOff
  Warning  Failed     47s (x2 over 75s)  kubelet, kube-master  Error: ErrImagePull
  Normal   BackOff    47s                kubelet, kube-master  Back-off pulling image "networkservicemesh/nsm-monitor:latest"
  Normal   Pulling    34s (x2 over 75s)  kubelet, kube-master  Pulling image "networkservicemesh/nsm-monitor:latest"
  Normal   BackOff    34s                kubelet, kube-master  Back-off pulling image "networkservicemesh/nsm-coredns:latest"
  Warning  Failed     34s                kubelet, kube-master  Error: ImagePullBackOff
```

I changed all the lastest tags in the code to master and saw all these problems registered in issues but I couldn't solve them.
Is there any way to solve this problem?

**Comment by edwarnicke** [comment]
@wjstk16 Thanks for bringing this to our attention.  nsm-coredns is new (hasn't hit a release) and it appears that we are not yet publishing it properly (apologies).  It also seems like our CI images are being put in the release repo (also not good).

We'll get it fixed shortly.  Thank you so much for speaking up and filing an issue.

**Comment by edwarnicke** [comment]
@wjstk16 I was looking at our CI repo when I wrote the above, should have looked at our release repo: https://hub.docker.com/r/networkservicemesh/nsm-coredns/tags

Because nsm-coredns has not been part of a release yet, we only have the branch label for 'master' (think latest on the master branch there).  Once we do a released version, it will get a latest.

All of that said... how can we make this less confusing?  If its confused you, it will confuse others...

**Comment by rdimitrov** [comment]
Related to https://github.com/networkservicemesh/networkservicemesh/issues/1551

**Comment by haiodo** [comment]
Fixed in #1560
